mws/apis/products.py

```diff
@@ -163,9 +163,89 @@ def get_lowest_priced_offers_for_asin(
         }
         return self.make_request(data)
 
-    # # # TODO add this
-    # def get_my_fees_estimate(self):
-    #     pass
+    def get_my_fees_estimate(
+        self, marketplace_id, id_type, id_value, is_amazon_fulfilled, identifier, listing_price, listing_price_currency_code
+    ):
+
+        """
+        Each argument must be a list, up to a length of 20, with id_value (number of ASINs or SellerSKUs) determining how many requests are made.
```

Comment:
This design gets cumbersome to manage, as it requires every argument to be a list of exactly 1 or exactly `n` elements, where `n` is supposed to match the length of one of those list arguments (`id_value`). This cannot easily account for situations where any given list contains `x` number of elements where `x != 1 and x != n`.

Further, we can certainly account for the 20 request limit specified in Amazon's documentation, but current code is not performing that check.

I think we'd be better served if each item is defined by a dictionary, like so:

```python
fee_requests = [
    {
        'id_value': 'B082YTKC47',
        'id_type': 'ASIN',
        'marketplace': 'A1F83G8C2ARO7P',
        'is_amazon_fulfilled': True,
        'identifier': 'Fee requests',
        'listing_price': {
            'amount': 19.0,
            'currency_code': 'GBP',
        },
        'shipping': {
            'amount': 5.0,
            'currency_code': 'USD',
        },
        'points': {
            'number': 5,
            'monetary_value': {
                'amount': 2.0,
                'currency_code': 'EUR',
            },
        }
    },
    # and so on, up to 20
]
```

The request method should then be able to accept one of the following signatures:

```python
# a single list argument, containing a list of dicts:
resp = products_api.get_my_fees_estimate(fee_requests=fee_requests)

# OR an arbitrary set of arguments that can be unpacked, each one being a dict:
resp = products_api.get_my_fees_estimate(*fee_requests)
```

I'm a little partial the second one myself, but either will do. 🤷

Reply:
Note that the example also includes `shipping` and `points`, which can also be requested according to documentation (see [PriceToEstimateFees](https://docs.developer.amazonservices.com/en_US/products/Products_Datatypes.html#PriceToEstimateFees) in the Products Datatypes doc). Full coverage of the request's potential parameters will be necessary before the method can be accepted.